doc-content/enterprise-only/openshift/environment-immutable-con.adoc

```diff
@@ -12,10 +12,14 @@ Optionally, you can also deploy a pod with {CENTRAL} Monitoring and a pod with S
 Smart Router is a single endpoint that can receive calls from client applications to any of your services and route each call automatically to the server that actually runs the process.
 
 endif::PAM[]
-When you create a {KIE_SERVER} image, you must build your services using S2I (Source to Image). Provide a Git repository with the source of your services and other business assets; if you develop the services or assets in {CENTRAL}, copy the source into a separate repository for the S2I build. OpenShift automatically builds the source, installs the process into the {KIE_SERVER} image, and starts the process. No further management of the image is required. If you want to use a new version of the process, you can build a new image.
+When you create a {KIE_SERVER} image, you can build your services using S2I (Source to Image). Provide a Git repository with the source of your services and other business assets; if you develop the services or assets in {CENTRAL}, copy the source into a separate repository for the S2I build. OpenShift automatically builds the source, installs the process into the {KIE_SERVER} image, and starts the process. 
 
 If you are using {CENTRAL} for authoring services, you can extract the source for your process and place it into a separate Git repository (such as GitHub or an on-premise installation of GitLab) for use in the S2I build.
 
+Alternatively, you can create an immutable {KIE_SERVER} image using services that are already built as KJAR files. In this case, you must provide the services in a Maven repository; you can use the built-in repository of the {CENTRAL} or your own repository (for example, a Nexus deployment). In an immutable deployment, the KJAR files are retrieved during the building of the image and not updated or changed after that. 
```

Comment:
It is a bit different here. In case you use "immutable image" with existing kjar then the kjar is downloaded when Kie server pod is started, **there is no building of the image**. Immutability means here that you cannot modify containers in running Kie server pods.